sonic_platform_base/sensor_base.py

```diff
@@ -0,0 +1,172 @@
+"""
+    sensor_base.py
+
+    Abstract base class for implementing a platform-specific class with which
+    to interact with a sensor module in SONiC
+"""
+
+from . import device_base
+
+SENSOR_TYPE_VOLTAGE = 1
+SENSOR_TYPE_CURRENT = 2
+
+class SensorBase(device_base.DeviceBase):
+    """
+    Abstract base class for interfacing with a sensor module
+    """
+
+    def get_type(self):
+        """
+        Specifies the type of the sensor.
+
+        Returns:
+            Sensor type
+        """
+        raise NotImplementedError
+
+    def get_value(self):
+        """
+        Retrieves measurement reported by sensor
+
+        Returns:
+            Sensor measurement
+        """
+        raise NotImplementedError
+
+    def get_unit(self):
+        """
+        Retrieves unit of measurement reported by sensor
+
+        Returns:
+            Sensor measurement unit
+        """
+        raise NotImplementedError
+
+    def get_high_threshold(self):
+        """
+        Retrieves the high threshold of sensor
+
+        Returns:
+            High threshold 
+        """
+        raise NotImplementedError
+
+    def get_low_threshold(self):
+        """
+        Retrieves the low threshold 
+
+        Returns:
+            Low threshold 
+        """
+        raise NotImplementedError
+
+    def set_high_threshold(self, value):
+        """
+        Sets the high threshold value of sensor
+
+        Args :
```

Comment:
No space before `:`, applies to all docstrings

Reply:
y